wdl/src/main/scala/wdl/WdlCall.scala

```diff
@@ -82,17 +83,21 @@ object WdlCall {
       *   input3 = other_task.out + 2   -> ExpressionNode with an input port pointing to the output port of other_task.out
       * }
      */
-    def expressionNodeMappings: ErrorOr[Map[LocalName, AnonymousExpressionNode]] = wdlCall.inputMappings traverse {
-      case (inputName, wdlExpression) =>
-        val identifier = wdlCall.womIdentifier.combine(inputName)
-        val constructor = wdlCall match {
-          case _: WdlTaskCall => TaskCallInputExpressionNode.apply _
-          case _ => PlainAnonymousExpressionNode.apply _
-        }
-
-        WdlWomExpression.toAnonymousExpressionNode(identifier, WdlWomExpression(wdlExpression, wdlCall), localLookup, outerLookup, preserveIndexForOuterLookups, wdlCall, constructor) map {
-          LocalName(inputName) -> _
-        }
+    def expressionNodeMappings: ErrorOr[Map[LocalName, AnonymousExpressionNode]] = {
+      val precomputedOgins: Map[String, OutputPort] = outerLookup.filterNot { ol => localLookup.contains(ol._1) } map { case (name, port) => name -> OuterGraphInputNode(WomIdentifier(name), port, preserveIndexForOuterLookups).singleOutputPort }
```

Comment:
Someday we'll be as evolved as [these guys](https://github.com/broadinstitute/clio/pull/182).

For now:

`val precomputedOgins: Map[String, OutputPort] = outerLookup filterNot { case (name, _) => localLookup.contains(name) } map { case (name, port) => name -> OuterGraphInputNode(WomIdentifier(name), port, preserveIndexForOuterLookups).singleOutputPort }`

Or:

```scala
val precomputedOgins: Map[String, OutputPort] = outerLookup filterNot {
  case (name, _) => localLookup.contains(name)
} map {
  case (name, port) =>
    name -> OuterGraphInputNode(WomIdentifier(name), port, preserveIndexForOuterLookups).singleOutputPort
}
```